### PR TITLE
[PREVIEW]Feature/sscs 3606 appeal created notification reps

### PR DESCRIPTION
**Pull request by iguannago**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapper.java [hunk 1]
```diff
@@ -67,6 +79,16 @@ public SscsCaseData getOldSscsCaseData() {
         return responseWrapper.getOldSscsCaseData();
     }
 
+    @Override
+    public List<SubscriptionWithType> getSubscriptionsBasedOnNotificationType() {
+        List<SubscriptionWithType> subscriptionWithTypeList = new ArrayList<>();
+        subscriptionWithTypeList.add(new SubscriptionWithType(getAppellantSubscription(), APPELLANT));
+        if (SYA_APPEAL_CREATED_NOTIFICATION.getId().equals(getNotificationType().getId())) {
+            subscriptionWithTypeList.add(new SubscriptionWithType(getRepresentativeSubscription(), REPRESENTATIVE));
```
Comment (sathishbabuj) [inline src/main/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapper.java]: Representative subscription may not exist for all appeals. So a check should be made for representative.

File: src/main/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapper.java [hunk 2]
```diff
@@ -67,6 +79,16 @@ public SscsCaseData getOldSscsCaseData() {
         return responseWrapper.getOldSscsCaseData();
     }
 
+    @Override
+    public List<SubscriptionWithType> getSubscriptionsBasedOnNotificationType() {
+        List<SubscriptionWithType> subscriptionWithTypeList = new ArrayList<>();
+        subscriptionWithTypeList.add(new SubscriptionWithType(getAppellantSubscription(), APPELLANT));
+        if (SYA_APPEAL_CREATED_NOTIFICATION.getId().equals(getNotificationType().getId())) {
```
Comment (sathishbabuj) [inline src/main/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapper.java]: Not sure why the representative subscription is created based on Id. It should be based on whether the representative subscription exists and is subscribed for email or sms notification

File: src/main/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapper.java [hunk 3]
```diff
@@ -32,6 +39,11 @@ public Subscription getAppellantSubscription() {
         return responseWrapper.getNewSscsCaseData().getSubscriptions().getAppellantSubscription();
     }
 
+    @Override
+    public Subscription getRepresentativeSubscription() {
+        return responseWrapper.getNewSscsCaseData().getSubscriptions().getRepresentativeSubscription();
```
Comment (sathishbabuj) [inline src/main/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapper.java]: A check should be added to fine whether representative subscription exists or not.

File: src/main/java/uk/gov/hmcts/reform/sscs/factory/NotificationFactory.java [hunk 4]
```diff
@@ -35,18 +44,28 @@
...
 
         return new Notification(template, destination, placeholders, reference, appealNumber);
     }
 
+    private Subscription getSubscriptionGivenSubscriptionType(SscsCaseData ccdResponse,
+                                                              SubscriptionType subscriptionType) {
+        if (APPELLANT.equals(subscriptionType)) {
+            return ccdResponse.getSubscriptions().getAppellantSubscription();
+        }
+        return ccdResponse.getSubscriptions().getRepresentativeSubscription();
```
Comment (sathishbabuj) [inline src/main/java/uk/gov/hmcts/reform/sscs/factory/NotificationFactory.java]: Need to add a check to verify representative subscription exists or not.

File: src/main/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapper.java (hunk 1 above)
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapper.java]: good catch, @sathishbabuj . I think it's not breaking the app as it creates even if it is null. But I am confirming it now and making sure it's covered by a junit test.

Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapper.java]: the subscription object is never null -> https://github.com/hmcts/sscs-track-your-appeal-notifications/blob/82ae894afb6c0da4cd558e340a6e6d10b22265cb/src/main/java/uk/gov/hmcts/reform/sscs/deserialize/SscsCaseDataWrapperDeserializer.java#L316

Anyway, added junit coverage here.

File: src/main/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapper.java (hunk 2 above)
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapper.java]: Yes, both subs are validated to confirm they are not null and there is some subscribed present either sms or email. But that is not done here. This method is just a wrapper to add the subscription Type: appellant or reps. Which needs to be passed down in the next methods.

File: src/main/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapper.java (hunk 3 above)
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapper.java]: as explained above the subscription object is always built even if in the json is not present.

File: src/main/java/uk/gov/hmcts/reform/sscs/personalisation/Personalisation.java [hunk 5]
```diff
@@ -231,10 +235,20 @@ private String formatLocalTime(LocalDateTime date) {
...
+        String emailTemplateName = getEmailTemplateName(subscriptionType, notificationWrapper.getNotificationType());
+        String smsTemplateName = isSendSmsSubscriptionConfirmation() ? SUBSCRIPTION_CREATED_NOTIFICATION.getId() :
+                emailTemplateName;
+        return config.getTemplate(emailTemplateName, smsTemplateName, benefit, notificationWrapper.getHearingType());
+    }
+
+    private String getEmailTemplateName(SubscriptionType subscriptionType,
+                                        NotificationEventType notificationEventType) {
+        String emailTemplateName = notificationEventType.getId();
+        if (SYA_APPEAL_CREATED_NOTIFICATION.getId().equals(notificationEventType.getId())) {
```
[the remark this answers is not in the document]
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/sscs/personalisation/Personalisation.java]: It is scalable as currently, it works for all types of notifications: paper, online, oral and regular. 
It is managed here -> https://github.com/hmcts/sscs-track-your-appeal-notifications/blob/eb1775b228de01f75ed765cc0fbb4cd03f54e12c/src/main/java/uk/gov/hmcts/reform/sscs/personalisation/Personalisation.java#L237

anyway, I improve the junit test covering this method by adding different test scenarios per HearingType to ensure it all works fine.

File: src/main/java/uk/gov/hmcts/reform/sscs/factory/NotificationFactory.java (hunk 4 above)
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/sscs/factory/NotificationFactory.java]: No need @sathishbabuj , as explained above the subscriptions is always built by the deserialiser.

File: src/main/java/uk/gov/hmcts/reform/sscs/personalisation/Personalisation.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/sscs/personalisation/Personalisation.java]: good point, @chrisgrimble , I change it.

File: src/main/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapper.java (hunk 1 above)
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapper.java]: fixed

**Pull request merged**